samba: update to 4.22.1

MingcongBai opened the pull request:
Topic Description
-----------------

- mplayer: backport from upstream to fix build
- gnome-vfs: bring in latest patch from Fedora
- ldb: separate from samba build
    Now newer Samba requires LDB > 2.9.0, but upstream LDB releases have stopped at version 2.9.2, which does not satisfy Samba runtime dependency.
    So we need to split LDB from the Samba build and package it.
- xine-lib: bump REL due to samba update to 4.22.1
- vlc: bump REL due to samba update to 4.22.1
- tdebase: bump REL due to samba update to 4.22.1
- sssd: bump REL due to samba update to 4.22.1
- pysmbc: bump REL due to samba update to 4.22.1
- pyside6: bump REL due to samba update to 4.22.1
- mplayer: bump REL due to samba update to 4.22.1

... and 13 more commits

Package(s) Affected
-------------------

- cifs-utils: 7.0-2
- elinks: 1:0.18.0-2
- ffmpeg: 7.1.1-1
- ffmpeg-static-libs-for-sunshine: 7.1.1-1
- gnome-control-center: 42.10-1
- gnome-vfs: 2.24.4-11
- gvfs: 1.50.2-2
- kio-extras: 23.08.5-1
- kodi: 1:21.2-1
- ldb: 2:4.22.1
- mplayer: 1:1.5-1
- pyside6: 6.8.2.1-1
- pysmbc: 1.0.25.1-1
- samba: 4.22.1
- sssd: 2.9.5-1
- talloc: 2.4.3
- tdb: 1.4.13
- tdebase: 14.1.2-2
- tevent: 0.16.2
- vlc: 3.0.21-7
- xine-lib: 1.2.13-5

Security Update?
----------------

No

Build Order
-----------

```
#buildit talloc tevent tdb samba:-pkgbreak cifs-utils elinks ffmpeg gnome-control-center gnome-vfs gvfs kio-extras kodi mplayer pyside6 pysmbc sssd tdebase vlc xine-lib samba
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

MingcongBai committed:
talloc: update to 2.4.3

MingcongBai committed:
tdb: update to 1.4.13

MingcongBai committed:
tevent: update to 0.16.2

MingcongBai committed:
samba: update to 4.22.1

MingcongBai committed:
samba: disable ldb as private library

- Improve prepare, build and beyond scripts.
- Add krb5 to PKGDEP.
- Update PKGBREAK.

CAB233 commented:
> Dickens-topic report has been uploaded to pastebin as [paste f61cafec-5ed1-4b26-9d63-35194f417be1](https://aosc.io/paste/detail?id=f61cafec-5ed1-4b26-9d63-35194f417be1).

SONAME changed.

CAB233 committed:
pyside6: bump REL due to samba update to 4.22.1

CAB233 committed:
kodi: bump REL due to samba update to 4.22.1

CAB233 committed:
mplayer: bump REL due to samba update to 4.22.1

CAB233 committed:
gnome-vfs: bump REL due to samba update to 4.22.1

CAB233 committed:
cifs-utils: bump REL due to samba update to 4.22.1

CAB233 committed:
ffmpeg: bump REL due to samba update to 4.22.1

CAB233 committed:
gnome-control-center: bump REL due to samba update to 4.22.1

CAB233 committed:
gvfs: bump REL due to samba update to 4.22.1

CAB233 committed:
kio-extras: bump REL due to samba update to 4.22.1

CAB233 committed:
elinks: bump REL due to samba update to 4.22.1

CAB233 committed:
pysmbc: bump REL due to samba update to 4.22.1

CAB233 committed:
vlc: bump REL due to samba update to 4.22.1

CAB233 committed:
sssd: bump REL due to samba update to 4.22.1

CAB233 committed:
tdebase: bump REL due to samba update to 4.22.1

CAB233 committed:
xine-lib: bump REL due to samba update to 4.22.1

CAB233 committed:
ldb: separate from samba build

Now newer Samba requires LDB > 2.9.0, but upstream LDB releases have stopped at version 2.9.2, which does not satisfy Samba runtime dependency.
So we need to split LDB from the Samba build and package it.

CAB233 committed:
gnome-vfs: bring in latest patch from Fedora

CAB233 committed:
mplayer: backport from upstream to fix build

CAB233 commented:
Tested on amd64
samba:
![](https://github.com/user-attachments/assets/4581c713-059f-4637-9775-5ab2f536cb6c)

elinks:
![](https://github.com/user-attachments/assets/21a055a5-7de4-4b24-9a38-375e3b5ec7c7)

ffmpeg:
![](https://github.com/user-attachments/assets/ae27d3e8-3e59-4f22-8171-464dab27cf43)

kodi:
![](https://github.com/user-attachments/assets/bbf69064-c9c5-4ef7-9c18-9212122cf215)

vlc:
![](https://github.com/user-attachments/assets/58018402-9786-4dd3-846d-07de2563d7b4)
